# DP: Don't add standard library dirs to library_dirs and runtime_library_dirs.

--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -213,6 +213,13 @@
                                         runtime_library_dirs)
         libraries, library_dirs, runtime_library_dirs = fixed_args
 
+        # filter out standard library paths, which are not explicitely needed
+        # for linking
+        library_dirs = [dir for dir in library_dirs
+                        if not dir in ('/lib', '/lib64', '/usr/lib', '/usr/lib64')]
+        runtime_library_dirs = [dir for dir in runtime_library_dirs
+                                if not dir in ('/lib', '/lib64', '/usr/lib', '/usr/lib64')]
+
         lib_opts = gen_lib_options(self, library_dirs, runtime_library_dirs,
                                    libraries)
         if not isinstance(output_dir, (str, type(None))):
